### ISSUE TITLE
Problem of the Softmax on Read Mode

### ISSUE THREAD

**Issue by yat011** [ixaxaar/pytorch-dnc#43]
https://github.com/ixaxaar/pytorch-dnc/blob/1db78511fe5622ade1c554d265a5d9d729c8801d/dnc/memory.py#L235

Should the softmax be applied on the last dimension? (i.e. the dimension of the read mode)

Currently, each read mode would always return 1 if the model has only one read head.

**Comment by ixaxaar** [comment]
That sounds logical, though lemme check the paper and get back on this.

**Comment by ixaxaar** [comment]
Hi @yat011 can you perhaps review the changes?

**Comment by yat011** [comment]
Hi @ixaxaar,
It looks good! Thanks a lot.
